src/prng.rs

```diff
@@ -7,51 +7,109 @@
 
 use super::field::{FieldElement, FieldError};
 use aes::{
-    cipher::{generic_array::GenericArray, FromBlockCipher, NewBlockCipher, StreamCipher},
+    cipher::{
+        generic_array::GenericArray, FromBlockCipher, NewBlockCipher,
+        StreamCipher as AesStreamCipher,
+    },
     Aes128, Aes128Ctr,
 };
 use getrandom::getrandom;
 
 use std::marker::PhantomData;
 
-const BLOCK_SIZE: usize = 16;
-const DEFAULT_BUFFER_SIZE_IN_ELEMENTS: usize = 128;
-const MAXIMUM_BUFFER_SIZE_IN_ELEMENTS: usize = 4096;
+const BUFFER_SIZE_IN_ELEMENTS: usize = 128;
 
-pub(crate) const SEED_LENGTH: usize = 2 * BLOCK_SIZE;
-
-pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, getrandom::Error> {
-    let mut seed = vec![0; SEED_LENGTH];
-    getrandom(seed.as_mut_slice())?;
+pub(crate) fn secret_share<F: FieldElement>(share1: &mut [F]) -> Result<Vec<u8>, PrngError> {
+    let mut seed = [0; Aes128Ctr::STREAM_CIPHER_SEED_SIZE];
+    getrandom(&mut seed)?;
 
     // get prng array
-    let data: Vec<F> = Prng::new_with_seed_and_optional_length(&seed, Some(share1.len())).collect();
+    let data: Vec<F> = Prng::<_, Aes128Ctr>::new_with_seed(&seed)?
+        .take(share1.len())
+        .collect();
 
     // secret share
     for (s1, d) in share1.iter_mut().zip(data.iter()) {
         *s1 -= *d;
     }
 
-    Ok(seed)
+    Ok(seed.to_vec())
 }
 
 pub(crate) fn extract_share_from_seed<F: FieldElement>(
     length: usize,
     seed: &[u8],
 ) -> Result<Vec<F>, PrngError> {
-    if seed.len() != 2 * BLOCK_SIZE {
-        return Err(PrngError::SeedLen);
+    Ok(Prng::<_, Aes128Ctr>::new_with_seed(seed)?
+        .take(length)
+        .collect())
+}
+
+/// Generate a vector of uniform random field elements.
+pub fn random_vector<F: FieldElement>(len: usize) -> Result<Vec<F>, PrngError> {
+    Ok(Prng::<_, Aes128Ctr>::new()?.take(len).collect())
+}
+
+/// A stream cipher maps a short key to a stream of pseudorandom bytes.
+pub(crate) trait StreamCipher: Sized {
+    const STREAM_CIPHER_SEED_SIZE: usize;
+
+    /// Returns a new stream ciphered keyed by `key`.
+    fn new(key: &[u8]) -> Result<Self, PrngError>;
```

Comment:
```suggestion
    fn new(key: &[u8; Self::STREAM_CIPHER_SEED_SIZE]) -> Result<Self, PrngError>;
```
Then you can skip the seed length checks in impls.

Reply:
If only :(
```
error: generic parameters may not be used in const operations
  --> src/prng.rs:61:25
   |
61 |     fn fella(key: &[u8; Self::STREAM_CIPHER_SEED_SIZE]) -> Self;
   |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot perform const operation using `Self`
   |
   = note: type parameters may not be used in const expressions

```

Reply:
Bummer! This is fine as-is then.